### PR TITLE
NCL-918: Centralise UI error reporting

### PR DESCRIPTION
**Pull request by alexcreasy**

### CODE REVIEW

**Comment by alexcreasy** [comment]
@vibe13 would you mind reviewing?

**Comment by vibe13** [comment]
@alexcreasy on it now

File: ui/app/configuration-set/configuration-set-controllers.js [hunk 1]
```diff
@@ -38,14 +35,10 @@
...
     'products',
     'PncRestClient',
-    'Notifications',
-    function($log, $state, products, PncRestClient, Notifications) {
-      $log.debug('ConfigurationSetCreateController >> this=%O, products=%O',
-                 this, products);
+    function($log, $state, products, PncRestClient) {
+      var self = this;
 
       this.data = new PncRestClient.ConfigurationSet();
```
Comment (vibe13) [inline ui/app/configuration-set/configuration-set-controllers.js]: self.data = new PncRestClient.ConfigurationSet();

File: ui/app/configuration/configuration-controllers.js [hunk 2]
```diff
@@ -27,25 +27,21 @@
...
-              if (result) {
-                self.projects.push(result);
-                //console.log(JSON.stringify(self.projects));
-              }
+      var that = this;
+
+      this.configurations = configurationList;
+      this.projects = [];
+
+      angular.forEach(this.configurations, function(configuration) {
```
Comment (vibe13) [inline ui/app/configuration/configuration-controllers.js]: To be consistent with other controllers, let's put:

var self = this;

self.configurations = configurationList;
self.projects = [];

angular.forEach(self.configurations, function(configuration) {

File: ui/app/configuration/configuration-controllers.js [hunk 3]
```diff
@@ -232,58 +204,29 @@
...
-
-          $state.go('configuration.detail.show', { configurationId: result.id });
-          Notifications.success('Configuration cloned');
-        },
-        function(response) {
-          $log.error('Clone configuration: %O failed, response: %O',
-                     that.configuration, response);
-          Notifications.error('Configuration clone failed');
-        });
+        this.configuration.$clone();
```
Comment (vibe13) [inline ui/app/configuration/configuration-controllers.js]: no more $state.go('configuration.detail.show', { configurationId: result.id });?

File: ui/app/services.js [hunk 4]
```diff
@@ -94,23 +96,47 @@
...
+          }
+        }
+
+      };
+    }
+  ]);
 
+  app.factory('httpResponseInterceptor', [
+    '$log',
+    '$q',
```
Comment (vibe13) [inline ui/app/services.js]: $q not required anymore here?

**Comment by vibe13** [comment]
@alexcreasy really cool refactoring, +1 for merging (minor comments above). Works as expected (tested creation/update/deletion/build). We should prioritize the messages creation on the REST APIs as now the Notification seen are quite useless, as expected. Thanks!

File: ui/app/configuration/configuration-controllers.js (hunk 2 above)
Comment (alexcreasy) [inline ui/app/configuration/configuration-controllers.js]: Actually we have a mix throughout the codebase. If you look at configuration-controllers.js it's entirely var that = this (the JS standard). 

Personally I prefer using the `this` keyword everywhere accept where it's necessary to use the saved reference (i.e. for functions that will be used as callbacks).

I'm happy to standardise this but there are a lot of places in the codebase that use the different methods. What do you think?

Comment (vibe13) [inline ui/app/configuration/configuration-controllers.js]: yes I agree, let's get rid of them unless necessary, but we can do it as we keep working on code, no need to do anything here

Comment (alexcreasy) [inline ui/app/configuration/configuration-controllers.js]: OK great,

**Comment by alexcreasy** [comment]
Retest this please

**Comment by alexcreasy** [comment]
@vibe13 would you mind doing a re-review. I think is is ready to merge if we're happy to lose the existing error / success messages.

**Comment by alexcreasy** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
+1 let's merge it!

**Pull request merged**